No Offerings return: Didn't find class "kotlinx.android.parcel.Parceler"

Affected version: 3.4.1.3

After trying to fetch the offerings, it returns successfully but returns zero offerings. It didn't crash but I noticed these messages in the log:

 ```
Rejecting re-init on previously-failed class java.lang.Class<com.revenuecat.purchases.parceler.JSONObjectParceler>: java.lang.NoClassDefFoundError: Failed resolution of: Lkotlinx/android/parcel/Parceler;
   at com.revenuecat.purchases.PurchaserInfo com.revenuecat.purchases.FactoriesKt.buildPurchaserInfo(org.json.JSONObject) (Factories.kt:60)
   at void com.revenuecat.purchases.Backend$getPurchaserInfo$call$1.onCompletion(com.revenuecat.purchases.HTTPClient$Result) (Backend.kt:89)
   at void com.revenuecat.purchases.Dispatcher$AsyncCall.run() (Dispatcher.kt:25)
   at void java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) (ThreadPoolExecutor.java:1167)
   at void java.util.concurrent.ThreadPoolExecutor$Worker.run() (ThreadPoolExecutor.java:641)
   at void java.lang.Thread.run() (Thread.java:764)
 Caused by: java.lang.ClassNotFoundException: Didn't find class "kotlinx.android.parcel.Parceler" on path: DexPathList[[zip file "/data/app/com.xx.xx-PZJTQZ0dFvYjs1quR7G-2Q==/base.apk"],nativeLibraryDirectories=[/data/app/com.xx.xx-PZJTQZ0dFvYjs1quR7G-2Q==/lib/arm, /data/app/com.xx.xx-PZJTQZ0dFvYjs1quR7G-2Q==/base.apk!/lib/armeabi-v7a, /system/lib]]
   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:134)
   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
   at com.revenuecat.purchases.PurchaserInfo com.revenuecat.purchases.FactoriesKt.buildPurchaserInfo(org.json.JSONObject) (Factories.kt:60)
   at void com.revenuecat.purchases.Backend$getPurchaserInfo$call$1.onCompletion(com.revenuecat.purchases.HTTPClient$Result) (Backend.kt:89)
   at void com.revenuecat.purchases.Dispatcher$AsyncCall.run() (Dispatcher.kt:25)
   at void java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) (ThreadPoolExecutor.java:1167)
   at void java.util.concurrent.ThreadPoolExecutor$Worker.run() (ThreadPoolExecutor.java:641)
   at void java.lang.Thread.run() (Thread.java:764)
```

The equivalent REST API call does return the correct results.

Please file a bug at https://github.com/thisisthekap/Xamarin.RevenueCat.Android

Sent from my iPhone

> On 22.09.2020, at 13:39, shimfish <notifications@github.com> wrote:
> 
> ﻿
> After trying to fetch the offerings, it returns successfully but returns zero offerings. Though this could be for a number of reasons, I noticed these messages in the log:
> 
> Rejecting re-init on previously-failed class java.lang.Class<com.revenuecat.purchases.parceler.JSONObjectParceler>: java.lang.NoClassDefFoundError: Failed resolution of: Lkotlinx/android/parcel/Parceler;
>   at com.revenuecat.purchases.PurchaserInfo com.revenuecat.purchases.FactoriesKt.buildPurchaserInfo(org.json.JSONObject) (Factories.kt:60)
>   at void com.revenuecat.purchases.Backend$getPurchaserInfo$call$1.onCompletion(com.revenuecat.purchases.HTTPClient$Result) (Backend.kt:89)
>   at void com.revenuecat.purchases.Dispatcher$AsyncCall.run() (Dispatcher.kt:25)
>   at void java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) (ThreadPoolExecutor.java:1167)
>   at void java.util.concurrent.ThreadPoolExecutor$Worker.run() (ThreadPoolExecutor.java:641)
>   at void java.lang.Thread.run() (Thread.java:764)
> Caused by: java.lang.ClassNotFoundException: Didn't find class "kotlinx.android.parcel.Parceler" on path: DexPathList[[zip file "/data/app/com.xx.xx-PZJTQZ0dFvYjs1quR7G-2Q==/base.apk"],nativeLibraryDirectories=[/data/app/com.xx.xx-PZJTQZ0dFvYjs1quR7G-2Q==/lib/arm, /data/app/com.xx.xx-PZJTQZ0dFvYjs1quR7G-2Q==/base.apk!/lib/armeabi-v7a, /system/lib]]
>   at java.lang.Class dalvik.system.BaseDexClassLoader.findClass(java.lang.String) (BaseDexClassLoader.java:134)
>   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String, boolean) (ClassLoader.java:379)
>   at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String) (ClassLoader.java:312)
>   at com.revenuecat.purchases.PurchaserInfo com.revenuecat.purchases.FactoriesKt.buildPurchaserInfo(org.json.JSONObject) (Factories.kt:60)
>   at void com.revenuecat.purchases.Backend$getPurchaserInfo$call$1.onCompletion(com.revenuecat.purchases.HTTPClient$Result) (Backend.kt:89)
>   at void com.revenuecat.purchases.Dispatcher$AsyncCall.run() (Dispatcher.kt:25)
>   at void java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker) (ThreadPoolExecutor.java:1167)
>   at void java.util.concurrent.ThreadPoolExecutor$Worker.run() (ThreadPoolExecutor.java:641)
>   at void java.lang.Thread.run() (Thread.java:764)
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

That is the github bug report.

Shim 🐟


On Tue, 22 Sep 2020 at 15:26, thisisthekap <notifications@github.com> wrote:

> Please file a bug at
> https://github.com/thisisthekap/Xamarin.RevenueCat.Android
>
> Sent from my iPhone
>
> > On 22.09.2020, at 13:39, shimfish <notifications@github.com> wrote:
> >
> > ﻿
> > After trying to fetch the offerings, it returns successfully but returns
> zero offerings. Though this could be for a number of reasons, I noticed
> these messages in the log:
> >
> > Rejecting re-init on previously-failed class
> java.lang.Class<com.revenuecat.purchases.parceler.JSONObjectParceler>:
> java.lang.NoClassDefFoundError: Failed resolution of:
> Lkotlinx/android/parcel/Parceler;
> > at com.revenuecat.purchases.PurchaserInfo
> com.revenuecat.purchases.FactoriesKt.buildPurchaserInfo(org.json.JSONObject)
> (Factories.kt:60)
> > at void
> com.revenuecat.purchases.Backend$getPurchaserInfo$call$1.onCompletion(com.revenuecat.purchases.HTTPClient$Result)
> (Backend.kt:89)
> > at void com.revenuecat.purchases.Dispatcher$AsyncCall.run()
> (Dispatcher.kt:25)
> > at void
> java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker)
> (ThreadPoolExecutor.java:1167)
> > at void java.util.concurrent.ThreadPoolExecutor$Worker.run()
> (ThreadPoolExecutor.java:641)
> > at void java.lang.Thread.run() (Thread.java:764)
> > Caused by: java.lang.ClassNotFoundException: Didn't find class
> "kotlinx.android.parcel.Parceler" on path: DexPathList[[zip file
> "/data/app/com.xx.xx-PZJTQZ0dFvYjs1quR7G-2Q==/base.apk"],nativeLibraryDirectories=[/data/app/com.xx.xx-PZJTQZ0dFvYjs1quR7G-2Q==/lib/arm,
> /data/app/com.xx.xx-PZJTQZ0dFvYjs1quR7G-2Q==/base.apk!/lib/armeabi-v7a,
> /system/lib]]
> > at java.lang.Class
> dalvik.system.BaseDexClassLoader.findClass(java.lang.String)
> (BaseDexClassLoader.java:134)
> > at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String,
> boolean) (ClassLoader.java:379)
> > at java.lang.Class java.lang.ClassLoader.loadClass(java.lang.String)
> (ClassLoader.java:312)
> > at com.revenuecat.purchases.PurchaserInfo
> com.revenuecat.purchases.FactoriesKt.buildPurchaserInfo(org.json.JSONObject)
> (Factories.kt:60)
> > at void
> com.revenuecat.purchases.Backend$getPurchaserInfo$call$1.onCompletion(com.revenuecat.purchases.HTTPClient$Result)
> (Backend.kt:89)
> > at void com.revenuecat.purchases.Dispatcher$AsyncCall.run()
> (Dispatcher.kt:25)
> > at void
> java.util.concurrent.ThreadPoolExecutor.runWorker(java.util.concurrent.ThreadPoolExecutor$Worker)
> (ThreadPoolExecutor.java:1167)
> > at void java.util.concurrent.ThreadPoolExecutor$Worker.run()
> (ThreadPoolExecutor.java:641)
> > at void java.lang.Thread.run() (Thread.java:764)
> > —
> > You are receiving this because you are subscribed to this thread.
> > Reply to this email directly, view it on GitHub, or unsubscribe.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/thisisthekap/Xamarin.RevenueCat.Android/issues/14#issuecomment-696687480>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKKC7PYPSQZ7JUSVQRUT3XTSHCJWXANCNFSM4RVVHR2Q>
> .
>

--- a/README.md
+++ b/README.md
@@ -31,6 +31,8 @@ If you encounter errors like `Java.Lang.NoClassDefFoundError: Failed resolution 
 	* Version >= 1.1.0
 * Xamarin.Android.Google.BillingClient
 	* Version >= 3.0.0
+* Karamunting.Kotlin.Android.Extensions.Runtime
+	* Version >= 1.3.72
 
 ## Xamarin.RevenueCat.Android.UsageChecker.UITests
 

--- a/README.md
+++ b/README.md
@@ -38,6 +38,16 @@ If you encounter errors like `Java.Lang.NoClassDefFoundError: Failed resolution 
 
 Use this project to briefly check if the bindings are working. Execute the Project in Visual Studio to execute the UI tests.
 
+## List runtime dependencies of `com.revenuecat.purchases:purchases`
+
+When creating the bindings for a new version, you might need to know the runtime classpath dependencies of `com.revenuecat.purchases:purchases`.
+
+Within the checked out RevenueCat repository, this command may be used to accomplish this task:
+
+`‌gradle dependencies --configuration releaseRuntimeClasspath`
+
+Optionally add the `--scan` option to the command to view a "web-based, searchable dependency report".
+
 ## License
 
 The license for this repository is specified in 

--- a/Xamarin.RevenueCat.Android.UsageChecker/Xamarin.RevenueCat.Android.UsageChecker.csproj
+++ b/Xamarin.RevenueCat.Android.UsageChecker/Xamarin.RevenueCat.Android.UsageChecker.csproj
@@ -121,6 +121,9 @@
         <PackageReference Include="Xamarin.Android.Google.BillingClient">
           <Version>3.0.0</Version>
         </PackageReference>
+        <PackageReference Include="Karamunting.Kotlin.Android.Extensions.Runtime">
+          <Version>1.3.72</Version>
+        </PackageReference>
     </ItemGroup>
     <ItemGroup>
         <PackageReference Include="Xamarin.AndroidX.Lifecycle.LiveData" Version="2.2.0.2" />

--- a/Xamarin.RevenueCat.Android/Xamarin.RevenueCat.Android.csproj
+++ b/Xamarin.RevenueCat.Android/Xamarin.RevenueCat.Android.csproj
@@ -3,7 +3,7 @@
     <PropertyGroup>
         <IsPackable>true</IsPackable>
         <PackageId>Xamarin.RevenueCat.Android</PackageId>
-        <Version>3.4.1.3</Version>
+        <Version>3.4.1.4-beta1</Version>
         <Authors>Christian Kapplmüller</Authors>
         <Company>fun.music IT GmbH</Company>
         <PackageOutputPath>nugetoutput</PackageOutputPath>
@@ -73,10 +73,11 @@
     </ItemGroup>
     <ItemGroup>
         <PackageReference Include="NuGet.Build.Tasks.Pack" Version="5.7.0" />
-        <PackageReference Include="Xamarin.Android.Google.BillingClient" Version="3.0.0" />
         <PackageReference Include="Xamarin.AndroidX.Annotation" Version="1.1.0" />
-        <PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime" Version="2.1.0" />
         <PackageReference Include="Xamarin.AndroidX.Lifecycle.Extensions" Version="2.1.0" />
+        <PackageReference Include="Xamarin.AndroidX.Lifecycle.Runtime" Version="2.1.0" />
+        <PackageReference Include="Xamarin.Android.Google.BillingClient" Version="3.0.0" />
+        <PackageReference Include="Karamunting.Kotlin.Android.Extensions.Runtime" Version="1.3.72" />
         <PackageReference Include="Xamarin.Kotlin.StdLib.Jdk7" Version="1.3.72" />
     </ItemGroup>
 

